unison-runtime/src/Unison/Runtime/Machine.hs

```diff
@@ -1833,23 +1879,26 @@ yield !env !denv !activeThreads !ustk !bstk !k = leap denv k
       ustk <- adjustArgs ustk ua
       bstk <- adjustArgs bstk ba
       apply env denv activeThreads ustk bstk k False (BArg1 0) clo
-    leap !denv (Push ufsz bfsz uasz basz rComb k) = do
-      let Lam _ _ uf bf nx = unRComb rComb
-      ustk <- restoreFrame ustk ufsz uasz
-      bstk <- restoreFrame bstk bfsz basz
-      ustk <- ensure ustk uf
-      bstk <- ensure bstk bf
-      eval env denv activeThreads ustk bstk k (rCombRef rComb) nx
+    leap !denv (Push ufsz bfsz uasz basz (CIx ref _ _) rComb k) = do
+      case unRComb rComb of
+        Lam _ _ uf bf nx -> do
+          ustk <- restoreFrame ustk ufsz uasz
+          bstk <- restoreFrame bstk bfsz basz
+          ustk <- ensure ustk uf
+          bstk <- ensure bstk bf
+          eval env denv activeThreads ustk bstk k ref nx
+        CachedClosure _w _clo -> do
+          error "TODO: Get help from Dan"
```

Comment:
Just need a hand from @dolio  on this bit, surprisingly all my tests so far have run without it 😆

Reply:
I refactored `MCode` `Let` and associated stuff so that this case no longer occurs. The combinator in a `Let` was meant to be the resumption to the middle of a function, so it makes no sense to have a cached value there. The combinator was only useful for doing a second stack protection check in case we're resuming via a captured continuation.

Might even speed things up a bit due to less indirection.